examples/mnist_tfrecord.py

```diff
@@ -0,0 +1,207 @@
+    img.set_shape([28 * 28])
+    img = tf.reshape(img, [28, 28, 1])
+
+    img = tf.cast(img, tf.float32) * (1. / 255) - 0.5
+
+    label = tf.cast(features['label'], tf.int32)
+    if one_hot and classes:
+        label = tf.one_hot(label, classes)
+
+    x_train_batch, y_train_batch = K.tf.train.shuffle_batch(
+        [img, label],
+        batch_size=batch_size,
+        capacity=2000,
+        min_after_dequeue=1000,
+        num_threads=32)  # set the number of threads here
+
+    return x_train_batch, y_train_batch
+
+
+def read_and_decode_recordinput(tf_glob, one_hot=True, classes=None, is_train=None, batch_size=1000):
+    """ Return tensor to read from TFRecord """
+
+    record_input = data_flow_ops.RecordInput(tf_glob, batch_size=batch_size)
+    records_op = record_input.get_yield_op()
+    records_op = tf.split(records_op, batch_size, 0)
+    records_op = [tf.reshape(record, []) for record in records_op]
+
+    imgs = []
+    labels = []
+    for serialized_example in records_op:
+        features = tf.parse_single_example(serialized_example,
+                                           features={
+                                               'label': tf.FixedLenFeature([], tf.int64),
+                                               'image_raw': tf.FixedLenFeature([], tf.string),
+                                           })
+        img = tf.decode_raw(features['image_raw'], tf.uint8)
+        img.set_shape([28 * 28])
+        img = tf.reshape(img, [1, 28, 28, 1])
+
+        img = tf.cast(img, tf.float32) * (1. / 255) - 0.5
+
+        label = tf.cast(features['label'], tf.int32)
+        if one_hot and classes:
+            label = tf.one_hot(label, classes)
+
+        imgs.append(img)
+        labels.append(label)
+
+    imgs = tf.concat(imgs, 0)
+    labels = tf.concat(labels, 0)
+    return imgs, labels
+
+
+def save_mnist_as_tfrecord():
+    (X_train, y_train), (X_test, y_test) = mnist.load_data()
+    X_train = X_train[..., np.newaxis]
+    X_test = X_test[..., np.newaxis]
+    images_to_tfrecord(images=X_train, labels=y_train, filename='train.mnist.tfrecord')
+    images_to_tfrecord(images=X_test, labels=y_test, filename='test.mnist.tfrecord')
+
+
+def cnn_layers(x_train_input):
+    con1 = Conv2D(32, (3, 3), activation='relu',
+                  strides=(2, 2), padding='valid')(x_train_input)
+    con2 = Conv2D(32, (3, 3), activation='relu',
+                  strides=(2, 2))(con1)
+    fla1 = Flatten()(con2)
+    den1 = Dense(128, activation='relu')(fla1)
+    x_train_out = Dense(classes,
+                        activation='softmax',
+                        name='x_train_out')(den1)
+    return x_train_out
+
+
+def create_cnn_model(x_train_batch, y_train_batch, x_batch_shape, y_batch_shape):
+    x_train_input = Input(tensor=x_train_batch, batch_shape=x_batch_shape)
+    x_train_out = cnn_layers(x_train_input)
+    # Workaround until _is_placeholder can be deduced automatically
+    x_train_out._is_placeholder = False
```

Comment:
This might also be addressed by https://github.com/fchollet/keras/pull/6928/files#r121752747